util/test/execution_limiter.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import with_statement
+
+import getpass
+import os
+import sys
+import tempfile
+import time
+
+def log_info(msg):
+    if os.getenv('CHPL_TEST_DEBUG_RUNNING_EXECUTABLE_LOCK') is not None:
+        sys.stdout.write('[Exec-lock Info: {0}]\n'.format(msg))
+        sys.stdout.flush()
+
+# Empty class that can be used with a context manager. Does not
+# limit how many executables can run at once.
+class NoLock():
+    def __enter__(self):
+        pass
+    def __exit__(self, type, value, traceback):
+        pass
+
+# Class that uses a file as a mutex so that only one executable can
+# run per user, per machine. This is useful when you want to
+# oversubscribe paratest, but don't want to cause too many
+# executables to run at once since some tests can consume the whole
+# machine which can starve other executables, causing timeouts.
+#
+# This works by using a file as a lock/mutex. It's race-y so it's
+# possible (though relatively unlikely) that two or more executables
+# could run at the same time. That's a little unfortunate, but not a
+# huge deal as it's pretty unlikely we'll get multiple resource
+# heavy tests running simultaneously
+#
+# The constructor takes the name of the executable and its timeout
+# value.
+class FileLock():
+    exec_name = ''
+    exec_timeout = ''
+    lock_file = ''
+
+    # simple backoff sleep, so that we have relatively fast latency for a bunch
+    # of quick running tests, but don't poll the machine continuously if a test
+    # is taking a while to run.
+    sleep_time = .1
+    def _backoff_sleep(self, max_sleep=.5, backoff_factor=1.1):
```

Comment:
Any rationale behind this backoff strategy? I assume you've tested and found that it works well?